spec/features/collection_spec.rb

```diff
@@ -53,6 +53,18 @@
       expect(page).to have_content(work1.title.first)
       expect(page).not_to have_content(work2.title.first)
     end
+
+    context "with a non-nestable collection type" do
+      let(:collection) do
+        create(:public_collection, user: user, description: ['collection description'], collection_type_settings: :not_nestable)
+      end
```

Comment:
Also, you can make this same test change in spec/features/dashboard/collection_spec.rb to check that the collection's admin show page handles this correctly.

Reply:
Thanks @elrayle. It doesn't look like members_count ever returns a nil, the methods that supply it always return an integer:

https://github.com/samvera/hyrax/blob/master/app/controllers/concerns/hyrax/collections_controller_behavior.rb#L79

https://github.com/samvera/hyrax/blob/master/app/controllers/hyrax/dashboard/collections_controller.rb#L437

So I didn't make an extra presenter method for that property.

I used the lw version in the collection spec, totally understandable why we'd prefer it, thanks for pointing the new factories out to me.

I could not get a dashboard spec to work though, and can't figure out why. I wrote a context and test and controller_type with the not-nested trait, but it failed with funny things at the very beginning of the test (can't find "New Collection", for example). Here's what I wrote that kept erroring:

```ruby
let(:collection_type_not_nested) { create(:collection_type, manager_user: admin_user, nestable: false, title: ['Not Nested']) }


context 'when collection type is not nestable'  do
  before do
    collection_type
    collection_type_not_nested

    sign_in user
    visit '/dashboard/my/collections'
  end

  it "makes a not-nested collection", :js do
    expect(page).to have_content("New Collection")
    click_link("New Collection")
    expect(page).to have_content 'Select type of collection'

    choose('Not Nested')
    click_on('Create collection')
  end
end
```

I did make sure the dashboard collection show uses the presenter method for subcollection_counts, and it doesn't error anymore, which is better, but not great without a test.

Reply:
If I'm reading this correctly, it looks like your are creating a collection type with title New Collection.  I think what you want here is...
```
let(:collection_not_nested) { create(:collection_lw, user: admin_user, collection_type_settings: [:not-nestable], title: ['Not Nested']) }
```

Reply:
Try this...
```
let(:collection_type_not_nested) { create(:collection_type, :not_nestable, manager_user: admin_user, title: ['Not Nested']) }
```
This uses the not_nestable trait.

Reply:
Just re-read you message to see what error you are seeing.  Can't see any reason why it wouldn't find the New Collection button.  I verified the case of the button label, because that can cause problems.

You might try have_button instead of have_content...
```
        expect(page).to have_button "New Collection"
```

Reply:
I tried that last night too, and tried varying the collection types I created in the setup, I am honestly surprised by these errors. I'll give a few more tries.